### ISSUE TITLE
Docker Install via Portainer

### ISSUE THREAD

**Issue by MexTrx** [MrBrax/TwitchAutomator#46]
Attempting to install this via docker via Portainer.io version 2.1.1 on my Synology NAS however when attempting to put the below in as a stack I get an error.

### Stack:
version: "3.8"
services:
  web:
    build: .
    image: mrbrax/twitchautomator
    ports:
      - "8082:8080"
    volumes:
      - ./volume2/docker/twitchautomator/config:/var/www/twitchautomator/config
      - ./volume2/docker/twitchautomator/cache:/var/www/twitchautomator/cache
      - ./volume2/docker/twitchautomator/logs:/var/www/twitchautomator/logs
      - ./volume2/docker/twitchautomator/vods:/var/www/twitchautomator/public/vods
      - ./volume2/docker/twitchautomator/saved_clips:/var/www/twitchautomator/public/saved_clips
      - ./volume2/docker/twitchautomator/saved_vods:/var/www/twitchautomator/public/saved_vods
  cron:
    build: ./docker/cron-container/
    image: mrbrax/twitchautomator:latest-cron
  broker:
    build: ./client-broker/
    image: mrbrax/twitchautomator:latest-broker

### Portainer Error: 
Deployment error
build path /data/compose/39/docker/cron-container either does not exist, is not accessible, or is not a valid URL.

Any help or maybe a better stack template someone has to run this?

**Comment by MrBrax** [comment]
are you running the build command from the same directory the project is in? have you tried pulling it from dockerhub instead?

**Comment by starkayc** [comment]
Running a similar setup to OP and also receive the same error. I tried pulling it from dockerhub and was able to run it but it returns with `nginx: [emerg] host not found in upstream "broker" in /etc/nginx/nginx.conf:86` while it's running. I also, tried running it via build command but received an error saying `ERROR: Version in "./docker-compose.yml" is unsupported. You might be seeing this error because you're using the wrong Compose file version`. 

My Docker version is 18.09.8 on Synology so I changed version to 3.7 and attempted to run it again but received an error saying `ERROR: build path /volume1/docker/twitchautomator/docker/cron-container either does not exist, is not accessible, or is not a valid URL.` So, I tried running the same docker pull command on my personal computer on Docker version 20.10.6 but also received a similar error saying build path `C:\Users\Starkayc\Desktop\docker\cron-container either does not exist, is not accessible, or is not a valid URL.`

**Comment by MrBrax** [comment]
is the cloned project folder directly on the desktop?

**Comment by starkayc** [comment]
Okay I finally figured out how to run everything and so far everything is running fine. Seems like once I downloaded `./docker/` and `./client-broker/` and `docker-compose.yml` into the same directory from this github and ran `docker-compose pull` and `docker-compose up -d` it all ran up and running. However, I was having permissions issues so whenever I would enter the dashboard it'd return HTTP 500 and logs would display it couldn't write to the directory. I was able to fix this using Portainer. I had to enable `Interactive & TTY (-i -t)` in Console Settings then deploy the container. Afterwards, I went to the console and for the command I selected /bin/ash and used root for the user. Once, inside I ran `chown -Rf nobody.nobody /var/www/twitchautomator/*` and everything worked as expected :D Also, what should the app url be? I have it at localhost:8080 but on the dashboard it shows disconnected on the bottom right but I'm still able to download vods normally. And I assume webhook would be localhost:8080/hook or something else? I assume it's that since after I added it the cron tab showed some curl stuff.
I'll list the steps incase anyone in the future needs help.

Steps Image: https://imgur.com/a/Zn93MOf

Steps: 
1. Have a folder for the docker. For me it's `/volume1/docker/twitchautomator`
2. Download the github code. Then extract the folder `client-broker` , `docker` and the `docker-compose.yml` file into your `twitchautomator` folder . So, the inside of `twitchautomator` folder would have `client-broker` , `docker` and the `docker-compose.yml`
3. SSH into your NAS and head to the folder you created for twitchautomator. Run `docker-compose pull` and it should be downloading 3 images and once that's done run `docker-compose up -d`
4. Everything should work in theory. If you received an error on the previous step due to version then edit your `docker-compose.yml` file and change the version from `3.8` to `3.7`.
5. Okay, so another error that you might receive is permissions. You'll know if your logs say `Saving config failed, permissions issue?`
6. Head to your Portainer Containers then click on the container that ends in web for me it's `twitchautomator_web_1` then click on `Duplicate/Edit` Button and Enable `Interactive & TTY (-i -t)` under the ` Advanced container settings ` tab. Once that's done click on the `Deploy the container`.
7. Now that the container is deployed with the new settings click on the container as before and now click on the `>_ Console` Button and for Command choose `/bin/ash` and for user choose `root` and then click on the `Connect` button. Once , you're inside run the command `chown -Rf nobody.nobody /var/www/twitchautomator/*` and everything should work. Be sure to restart the container and see if it works after boot as well.
8. Everything should be working fine now.

**Comment by MrBrax** [comment]
the app url has to be your external url, either your ip address or your domain name

the broker should handle the websocket messages from the backend to the frontend, probably set up incorrectly due to this unusual setup?

**Comment by starkayc** [comment]
Yeah after adding the app url and enabling websocket it seems to show connected now. Everything works fine from my testing other than the Render Chat/Render & Burn buttons in the dashboard. I can go to the tools tab and download chat but for vods I only got it to work for a vod that was like 30 mins if I tried another one that's longer like 4 hours I get "Request failed with status code 504". I also, get the error "Invalid API endpoint".

Settings: https://imgur.com/a/QkaNVhm
